src/components/v2/Layout/Header/index.tsx

```diff
@@ -19,19 +24,35 @@ const Header = () => {
   const isOnMobile = useIsMdDown();
 
   const title = useMemo(() => {
+    // Handle special case of Market Details page
+    if (isOnMarketDetailsPage(pathname)) {
+      const vTokenId = pathname.substring(pathname.lastIndexOf('/') + 1) as VTokenId;
+      const token = getToken(vTokenId);
+
+      return (
+        <Link to={Path.MARKET} css={styles.backButton}>
+          <Icon name="chevronLeft" css={styles.backButtonChevronIcon} />
+          <Icon name={vTokenId} css={styles.backButtonTokenIcon} />
+          <h3 css={styles.backButtonTokenSymbol}>{token.symbol}</h3>
+        </Link>
+      );
+    }
+
     const currentItem = menuItems.find(item => item.href === pathname);
     const currentItemKey = currentItem?.i18nTitleKey || currentItem?.i18nKey;
+
     if (!currentItemKey) {
-      console.error(`missed key for translation: ${pathname}`);
+      console.error(`Missing translation for key: ${pathname}`);
```

Comment:
We could leave a comment here to add error reporting which would help us in the future when we add better error reporting

Reply:
Good idea.